app/src/main/java/com/m2049r/xmrwallet/WalletFragment.java

```diff
@@ -141,9 +147,25 @@ public void onNothingSelected(AdapterView<?> parentView) {
         return view;
     }
 
+    void updateBalance() {
+        if (isExchanging) return; // wait for exchange to finish - it will fire this itself then.
+        // at this point selection is XMR in case of error
+        String displayB;
+        double amountA = Double.parseDouble(Wallet.getDisplayAmount(unlockedBalance)); // assume this cannot fail!
```

Comment:
The intended behaviour is that the app crashes if this happens because it means the monero code we are using is broken. Maybe this should be reflected in the comment instead of "assume".